hudi-common/src/main/java/org/apache/hudi/common/model/HoodieLogFile.java

```diff
@@ -42,63 +45,98 @@ public class HoodieLogFile implements Serializable {
 
   private static final Comparator<HoodieLogFile> LOG_FILE_COMPARATOR = new LogFileComparator();
   private static final Comparator<HoodieLogFile> LOG_FILE_COMPARATOR_REVERSED = new LogFileComparator().reversed();
+  // Log files are of this pattern - .b5068208-e1a4-11e6-bf01-fe55135034f3_20170101134598.log.1
+  public static final Pattern LOG_FILE_PATTERN =
+      Pattern.compile("\\.(.*)_(.*)\\.(.*)\\.([0-9]*)(_(([0-9]*)-([0-9]*)-([0-9]*)))?");
 
   private transient FileStatus fileStatus;
   private final String pathStr;
   private long fileLen;
+  private transient Path path;
+
+  private String baseCommitTime;
+  private int fileVersion;
+  private String logWriteToken;
+  private String fileExtension;
+  private String fileId;
```

Comment:
This will be very helpful to improve the performance when meet a lot of logfiles which need to resolve using RegEx. I am a little concerned if it will occupy more heap footprint. Anyway compare to memory overhead, the performance improvement is more considerable in my opinion.